tests/test_rest.py

```diff
@@ -101,3 +103,12 @@ def test_data(self, get):
         self.assertTrue(response.raise_for_status.called)
         self.assertEqual(get.call_args[0][0], 'http://127.0.0.1:8181/foo')
         self.assertEqual(get.call_args[1]['data'], json.dumps(data))
+
+    @patch('requests.get')
+    def test_get_text(self, get):
+        response = mock_response()
+        response.json = lambda: throw(ValueError)
```

Comment:
According to the docs for Mock, this is the way to do this:
response.json.side_effect = ValueError()

https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.side_effect